### PR TITLE
INT-376 Jenkinsfile to build and release

### PR DESCRIPTION
**Pull request by bigspotteddog**
https://issues.sonatype.org/browse/INT-376

Builds berks package, tests the package, creates tags and a release on github, then publishes the tar.gz artifact.

This is the screencast from INT-377, but it is the same process: https://youtu.be/hlVhhdx0Lh0

### CODE REVIEW

File: .kitchen.yml.erb [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+# Copyright:: Copyright (c) 2017-present Sonatype, Inc. Apache License, Version 2.0.
+driver:
+  name: ec2
+  aws_ssh_key_id: <%= ENV['KEY_PAIR_NAME'] %>
+  region: us-east-1
+  security_group_ids: ["sg-a4fc5ec1"]
+  require_chef_omnibus: true
+  subnet_id: subnet-c96f61bd
```
Comment (fernau) [inline .kitchen.yml.erb]: it is probably slightly preferable to create and delete the security group and subnet. if that is to much work add them as parameters to the Jenkinsfile.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        dir('build/target') {
+          OsTools.runSafe(this, "tar -zxvf ${archiveName}")
+        }
+
+        dir("build/target/cookbooks/${cookbookName}") {
+          OsTools.runSafe(this, "KEY_PAIR_NAME=${keyPairName} erb ${WORKSPACE}/.kitchen.yml.erb > .kitchen.yml")
+          OsTools.runSafe(this, 'cp ${WORKSPACE}/Berksfile .')
+          OsTools.runSafe(this, 'cp ${WORKSPACE}/Berksfile.lock .')
+          OsTools.runSafe(this, 'cp ${WORKSPACE}/metadata.rb .')
+          OsTools.runSafe(this, 'kitchen test')
```
Comment (fernau) [inline Jenkinsfile]: which account would this run under ?

Comment (bigspotteddog) [reply Jenkinsfile]: This is running under the jenkins user.

File: .kitchen.yml.erb (hunk 1 above)
Comment (bigspotteddog) [reply .kitchen.yml.erb]: Parameterized security group and subnet.

**Pull request merged**